ci/ray_ci/upgrade_version.py

```diff
@@ -0,0 +1,107 @@
+import click
+import os
+import subprocess
+from typing import List
+
+bazel_workspace_dir = os.environ.get("BUILD_WORKSPACE_DIRECTORY", "")
+
+non_java_files = [
+    "ci/ray_ci/utils.py",
+    "python/ray/_version.py",
+    "src/ray/common/constants.h",
+]
+
+MASTER_BRANCH_VERSION = "3.0.0.dev0"
+MASTER_BRANCH_JAVA_VERSION = "2.0.0-SNAPSHOT"
+
+
+def list_java_files(root_dir: str):
+    """
+    Scan the directories and return the sorted list of
+        pom.xml and pom_template.xml files.
+    """
+    files = []
+    for current_root_dir, _, file_names in os.walk(root_dir):
+        for file_name in file_names:
+            if file_name in ["pom.xml", "pom_template.xml"]:
+                files.append(os.path.join(current_root_dir, file_name))
+    return sorted(files)
+
+
+def get_check_output(file_path: str):
+    return subprocess.check_output(["python", file_path], text=True)
+
+
+def get_current_version(root_dir: str):
+    """
+    Scan for current Ray version and return the current versions.
+    """
+    version_file_path = os.path.join(root_dir, "python/ray/_version.py")
+    ray_version_output = get_check_output(version_file_path).split()
+    if len(ray_version_output) != 2:
+        raise ValueError(
+            f"Unexpected output from {version_file_path}: {ray_version_output}"
+        )
+    version = ray_version_output[0]
+
+    if version != MASTER_BRANCH_VERSION:
+        main_version = version
+        java_version = version
+        return main_version, java_version
+    return MASTER_BRANCH_VERSION, MASTER_BRANCH_JAVA_VERSION
+
+
+def upgrade_file_version(
+    non_java_files: List[str],
+    java_files: List[str],
```

Comment:
rather than sending the file list in, could you move the listing logic inside this function, so that it is nested in?

Reply:
sorry that I did not notice this in the last round of the review.

Reply:
done!